src/main/java/com/ericsson/ei/rules/RulesHandler.java

```diff
@@ -39,57 +45,105 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 
 @Component
-@Scope(value="thread", proxyMode = ScopedProxyMode.TARGET_CLASS)
+@Scope(value = "thread", proxyMode = ScopedProxyMode.TARGET_CLASS)
 public class RulesHandler {
     private static final Logger LOGGER = LoggerFactory.getLogger(RulesHandler.class);
 
-    @Value("${rules.path}") private String jsonFilePath;
+    private static final String EI_HOME_DEFAULT_NAME = ".eiffel-intelligence";
+
+    @Value("${rules.path}")
+    private String rulesFilePath;
 
     private JmesPathInterface jmesPathInterface = new JmesPathInterface();
     private static String jsonFileContent;
     private static JsonNode parsedJson;
-    
+
     public RulesHandler() {
         super();
     }
-    
+
     public void setParsedJson(String jsonContent) throws JsonProcessingException, IOException {
         ObjectMapper objectmapper = new ObjectMapper();
         parsedJson = objectmapper.readTree(jsonContent);
     }
+
+    public String readRulesFileContent(String path) throws Exception {
+        setRulesFilePath(path);
+        downloadRulesFile(path);
+        return readRulesJsonFileContent();
+    }
     
-    public String readRuleFileContent(String ruleFilePath) throws IOException {
-        String ruleJsonFileContent;
-        InputStream in = this.getClass().getResourceAsStream(ruleFilePath);
-        if(in == null) {
-            ruleJsonFileContent = FileUtils.readFileToString(new File(ruleFilePath), Charset.defaultCharset());
+    public void setRulesFilePath(String path) throws IOException {
```

Comment:
Missing javadoc on most methods in this class.

Reply:
Done.